Source/DafnyLanguageServer/Workspace/DiagnosticPublisher.cs

```diff
@@ -26,11 +29,39 @@ public class DiagnosticPublisher : IDiagnosticPublisher {
     }
 
     private void PublishDocumentDiagnostics(DafnyDocument document) {
-      languageServer.TextDocument.PublishDiagnostics(new PublishDiagnosticsParams {
+      var diagnosticParameters = new PublishDiagnosticsParams {
         Uri = document.Uri,
         Version = document.Version,
         Diagnostics = GetDiagnostics(document).ToArray(),
-      });
+      };
+      languageServer.TextDocument.PublishDiagnostics(diagnosticParameters);
+    }
+
+    public void PublishVerificationDiagnostics(DafnyDocument document) {
```

Comment:
Nitpick: consider moving this to `VerificationProgressReporter` since it is called there.

Reply:
I considered it, but since this file is the endpoint of all publications (notifications and diagnostics), I think it's better there than in `VerificationProgressReporter`.